packages/kit/src/runtime/client/client.js

```diff
@@ -1889,50 +1891,53 @@ async function load_data(url, invalid) {
 
 		let text = '';
 
-		while (true) {
-			// Format follows ndjson (each line is a JSON object) or regular JSON spec
-			const { done, value } = await reader.read();
-			if (done && !text) break;
+		async function handle_stream() {
+			while (true) {
+				// Format follows ndjson (each line is a JSON object) or regular JSON spec
+				const { done, value } = await reader.read();
+				if (done && !text) break;
 
-			text += !value && text ? '\n' : decoder.decode(value); // no value -> final chunk -> add a new line to trigger the last parse
+				text += !value && text ? '\n' : decoder.decode(value); // no value -> final chunk -> add a new line to trigger the last parse
 
-			while (true) {
-				const split = text.indexOf('\n');
-				if (split === -1) {
-					break;
-				}
+				while (true) {
+					const split = text.indexOf('\n');
+					if (split === -1) {
+						break;
+					}
 
-				const node = JSON.parse(text.slice(0, split));
-				text = text.slice(split + 1);
+					const node = JSON.parse(text.slice(0, split));
+					text = text.slice(split + 1);
 
-				if (node.type === 'redirect') {
-					return resolve(node);
-				}
+					if (node.type === 'redirect') {
+						return resolve(node);
+					}
 
-				if (node.type === 'data') {
-					// This is the first (and possibly only, if no pending promises) chunk
-					node.nodes?.forEach((/** @type {any} */ node) => {
-						if (node?.type === 'data') {
-							node.uses = deserialize_uses(node.uses);
-							node.data = deserialize(node.data);
-						}
-					});
+					if (node.type === 'data') {
+						// This is the first (and possibly only, if no pending promises) chunk
+						node.nodes?.forEach((/** @type {any} */ node) => {
+							if (node?.type === 'data') {
+								node.uses = deserialize_uses(node.uses);
+								node.data = deserialize(node.data);
+							}
+						});
 
-					resolve(node);
-				} else if (node.type === 'chunk') {
-					// This is a subsequent chunk containing deferred data
-					const { id, data, error } = node;
-					const deferred = /** @type {import('types').Deferred} */ (deferreds.get(id));
-					deferreds.delete(id);
+						resolve(node);
+					} else if (node.type === 'chunk') {
+						// This is a subsequent chunk containing deferred data
+						const { id, data, error } = node;
+						const deferred = /** @type {import('types').Deferred} */ (deferreds.get(id));
+						deferreds.delete(id);
 
-					if (error) {
-						deferred.reject(deserialize(error));
-					} else {
-						deferred.fulfil(deserialize(data));
+						if (error) {
+							deferred.reject(deserialize(error));
+						} else {
+							deferred.fulfil(deserialize(data));
+						}
 					}
 				}
 			}
 		}
+		handle_stream();
```

Comment:
What's this change about btw?

Reply:
ah I was trying to resolve to the TODO about the eslint error https://eslint.org/docs/latest/rules/no-async-promise-executor I can revert it though